python/lsst/sims/photUtils/BandpassDict.py

```diff
@@ -0,0 +1,331 @@
+import copy
+import numpy
+from collections import OrderedDict
+from lsst.sims.photUtils import Bandpass, Sed
+
+__all__ = ["BandpassDict"]
+
+class BandpassDict(object):
+    """
+    This class will wrap an OrderedDict of Bandpass instantiations.
+
+    Upon instantiation, this class's constructor will resample
+    the input Bandpasses to be on the same wavelength grid (defined
+    by the first input Bandpass).  The constructor will then calculate
+    the 2-D phiArray for quick calculation of magnitudes in all
+    Bandpasses simultaneously (see the member method calcMagListFromSed).
+    """
+
+    def __init__(self, bandpassList, bandpassNameList):
+        """
+        @param [in] bandpassList is a list of Bandpass instantiations
+
+        @param [in] bandpassNameList is a list of tags to be associated
+        with those Bandpasses
+        """
+        self._bandpassDict = OrderedDict()
+        for bandpassName, bandpass in zip(bandpassNameList, bandpassList):
+
+            if bandpassName in self._bandpassDict:
+                raise RuntimeError("The bandpass %s occurs twice in your input " % bandpassName \
+                                   + "to BandpassDict")
+
+            self._bandpassDict[bandpassName] = copy.deepcopy(bandpass)
+
+        dummySed = Sed()
+        self._phiArray, self._wavelenStep = dummySed.setupPhiArray(self._bandpassDict.values())
```

Comment:
I wonder if setUpPhiArray can be a method of something bandpass related, since phiarray is really associated with a bandpass. Maybe at some other time, there can be a function to sample a bandpass phiArray at a Sed's wavelengths.